app/controllers/fact_values_controller.rb

```diff
@@ -24,12 +25,30 @@ def index
       values = values.root_only
     end
 
-    @fact_values = values.no_timestamp_facts.required_fields.paginate :page => params[:page]
+    @fact_values = values.no_timestamp_facts
+
+    respond_to do |format|
+      format.html do
+        @fact_values = @fact_values.preload(related_tables).paginate :page => params[:page]
+        render :index
+      end
+      format.csv do
+        csv_response(@fact_values.joins(related_tables).includes(related_tables))
```

Comment:
`joins` is needed to force an inner join instead of a left outer join in the `includes`. When pulling in 1000 records at a time, using an inner join is faster then then preloading.